go-controller/hybrid-overlay/pkg/types/types.go

```diff
@@ -11,6 +11,8 @@ const (
 	HybridOverlayNodeSubnet = HybridOverlayAnnotationBase + "node-subnet"
 	// HybridOverlayDRMAC holds the MAC address of the Distributed Router/gateway
 	HybridOverlayDRMAC = HybridOverlayAnnotationBase + "distributed-router-gateway-mac"
+	// HybridOverlayIFAddress holds the interface address to redirect traffic to get to the hybrid overla
+	HybridOverlayIFAddress = HybridOverlayAnnotationBase + "interface-address"
```

Comment:
@dcbw can you weigh in on this, you suggested moving away from distributed router would be a good direction.

Reply:
I'm not sure interface address is better because there is no interface. The IP is just manipulated by openflow flows. I thought DR stood for designated router. It's definitely not a distributed thing.

Reply:
@trozet it's definitely "Distributed Router" but of *course* that's the not the same thing as OVN"s distributed router...

```
hybrid-overlay/pkg/controller/node_windows.go:          DistributedRouterMacAddress: drMAC.String(),
vendor/github.com/Microsoft/hcsshim/hcn/hcnpolicy.go:   DistributedRouterMacAddress string
```

Reply:
Forgive me if I'm not current on all the context...

The hybrid overlay port on each node switch still needs an IP address so we can send traffic to it. That address is currently the .3 address, but this PR is attempting to change that IP to be different on every node. The problem is that we need somewhere to store that address persistently.

(though maybe we don't? We could just reconfigure if we can't find out what it should be, but that seems lame. Also... Node Custom Resource anyone???)

So... stuffing it into a node annotation like the DRMAC was the natural path here, I think. Only the master knows what IPs are free since it handles the address assignment for the node's subnet, so it gets to pick a free address and it uses that to set up the LR Policy reroute. The node needs to know what that address is so it can do the br-ext OpenFlow parkour.

If we don't have an actual port on the node switch that "owns" that IP address, then we can't reroute traffic to it (and thus get rid of the in-pod routing table route), and thus we can't dynamically enable/disable hybrid overlay at runtime.

Reply:
BUT... this port itself doesn't have anything to do with a distributed router, since the DR thing is a Windows-only thing.

So we can literally call it whatever we want, and we currently call it the "Hybrid Overlay Port" eg `util.GetHybridOverlayPortName(n.nodeName)`.

So if we don't want to call it an interface address, maybe `HybridOverlayPortAddress` instead?

Reply:
I view the DR as the "router" implemented by openflow on br-ext. If I look at the code that comes up with the "DR MAC":
```
	// nothing allocated, allocate default mac
	if portMAC == nil && annotationMAC == nil {
		for _, subnet := range subnets {
			ip := util.GetNodeHybridOverlayIfAddr(subnet).IP
			portMAC = util.IPAddrToHWAddr(ip)
			annotationMAC = portMAC
			if !utilnet.IsIPv6(ip) {
				break
			}
		}
```

Here we can see the DR mac is the .3 IP converted to a mac address. Which tells me that the DR MAC is essentially the same "port" as the .3 IP. Since what my pod is sending to is this DR port, and this DR port essentially responds to ARP and acts as though it exists as br-ext, it is not a windows thing but is some router port emulated inside of br-ext on my node. Therefore I'm confused how we can call the mac that is derived from the .3 address a DR MAC, but we don't call the IP address a DR IP? We also call it dr.ip in the code...

I'm not advocating to keep the name "Distributed Router". The router is not distributed, it individually exists on each node logically and physically. I just don't think we can change the naming convention for our old annotation and I hesitate to come up with a new name for a new annotation. I think they should just stay consistent or else it will become even more confusing.

Reply:
@trozet @dcbw I changed it to `HybridOverlayPortAddress = HybridOverlayAnnotationBase + "port-addresses"` how does that sound to everyone?

Reply:
@JacobTanenbaum spoke with @dcbw and he is OK with just making it distributed-router-gateway-ip for consistency with the other annotation and code terminology

Reply:
changed it to `distributed-router-gateway-ip`